tardis/energy_input/base.py

```diff
@@ -0,0 +1,369 @@
+)
+from tardis.energy_input.gamma_ray_interactions import (
+    scatter_type,
+    compton_scatter,
+    pair_creation,
+    get_compton_angle,
+)
+from tardis.energy_input.util import (
+    get_random_theta_gamma_ray,
+    get_random_phi_gamma_ray,
+)
+from tardis import constants as const
+from astropy.coordinates import cartesian_to_spherical
+from tardis.montecarlo.montecarlo_numba.numba_config import CLOSE_LINE_THRESHOLD
+
+
+def initialize_packets(
+    number_of_shells,
+    packets_per_shell,
+    ejecta_volume,
+    inner_velocities,
+    outer_velocities,
+    decay_rad_db,
+):
+    """Initializes packet properties
+    and appends beta decay energy to output tables
+
+    Parameters
+    ----------
+    number_of_shells : int64
+        Number of shells in model
+    packets_per_shell : pandas dataframe
+        Number of packets in a shell
+    ejecta_volume : array of float64
+        Volume per shell
+    inner_velocities : array of float64
+        Shell inner velocities
+    outer_velocities : array of float64
+        Shell outer velocities
+    decay_rad_db : pandas dataframe
+        Decay radiation database
+
+    Returns
+    -------
+    list
+        GXPacket objects
+    numpy array
+        energy binned per shell
+    list
+        packet info
+    """
+    packets = []
+    energy_df_rows = np.zeros(number_of_shells)
+    energy_plot_df_rows = []
+
+    for column in packets_per_shell:
+        subtable = decay_rad_db.loc[column]
+        gamma_ratio, positron_ratio = intensity_ratio(
+            subtable, "'gamma_rays' or type=='x_rays'", "'e+'"
+        )
+        energy_sorted, energy_cdf = setup_input_energy(
+            subtable, "'gamma_rays' or type=='x_rays'"
+        )
+        positron_energy_sorted, positron_energy_cdf = setup_input_energy(
+            subtable, "'e+'"
+        )
+        for shell in range(number_of_shells):
+            required_packets_in_shell = packets_per_shell[column].iloc[shell]
+            for i_packet in range(required_packets_in_shell):
+                # draw a random gamma-ray in shell
+                ray1 = GXPacket(0, 0, 1, GXPacketStatus.IN_PROCESS, 0)
+                z1 = np.random.random()
+                initial_radius = inner_velocities[shell] + z1 * (
+                    outer_velocities[shell] - inner_velocities[shell]
+                )
+                location_theta = get_random_theta_gamma_ray()
+                location_phi = get_random_phi_gamma_ray()
+                ray1.location = SphericalVector(
+                    initial_radius, location_theta, location_phi
+                )
+                direction_theta = get_random_theta_gamma_ray()
+                direction_phi = get_random_phi_gamma_ray()
+                ray1.direction = SphericalVector(
+                    1.0, direction_theta, direction_phi
+                )
+                ray1.shell = shell
+                z2 = np.random.random()
+                if z2 > gamma_ratio:
```

Comment:
```suggestion
                if gamma_ratio < np.random.random():
```

Reply:
It would be useful to have a comment here explaining which physical scenario this corresponds to (positron annihilation ...).

Reply:
gamma_ratio is not the best name for this. How about 'gamma_ray_probability'?